_posts/2024-11-24-Bulk.markdown

```diff
@@ -27,3 +27,15 @@ Later in the week I set off on doing some site cleanup (added an icon for the br
 
+## End of Week 3 12/6/24
+Another week down. Time flies, huh? Well, we finished up our first iteration of the "Line Follower." And just for future reference, when I say we I mean my dad and I. He is a software engineer turned teacher. Not really, but he's been the one showing me the ropes. I am getting better at grasping the concepts, using git, and coding in general. He has been a huge help in all of those things. Anyway, the rudementary "motorturn" code block was a fun challenge. We had a black line made of electrical tape, and a color sensing robot on treads. We wanted it to follow the line as accurately as possible. It was a fun combo of software "engineering" as my dad calls it (the process of writing concise and well organized code which is easy to read and manipulate) and using critical thinking skills to work around the big elephant in the room:we had one color sensor. As one with basic coding knowledge would surely understand, the presence of multiple color sensors would allow for an easy solution. Something along the lines of, "if they both detect black, bingo! If not, well, rotate until they both detect black!" Not that the real process was much more complex, but yours truly is very new to robotics. Thankfully, I was able to put together some functional code and get it rolling (in case you were curious the final code is in my repository "starterlego"). 
+
+#### Big Takeaways
+-When I set to work to try to figure out the problem on my own, I discovered that a "range" function existed. This allows for a variable to be set to multiple numbers in a range, which made the color sensor much easier to use. You see, to set a target light intensity for the sensor to follow, I had to test the light intensity of the tape. The brace holding the sensor being less than perfect, it would be closer and farther away from the tape on a day to day basis. This of course changed the value of the intensity each time I tested it. With the range, I did not have to change the target intensity every time I ran the program. Convienent!
+
+-Software engineering is cool. The process of consilidating code is very satisfying. I defined various "def"s and set the values of the rotation angle, speed, etc. I would do this for each of the motors, which accounted for several unecessary lines of code. My dad showed me that I could simply make one def with rot angle, speed, etc. passed as variables. This would contain instructions to run the motors that could be called in the "def"s for left motor and right motor. Simply changing the values for speed, rot angle, etc. gives you a concise way of doing the same thing. I just did a very poor job of explaining this process, but these descriptions are usually from memory and all part of the learning process for me. Dad and I usually go over my mistakes later on. Who knows, you may even get my edited version and think me a prodigy!
+
+#### The Week
```

Comment:
"unfulling" - should that be "unfulfilling"?

I might capitalize Wall, maybe make a reference (and link) to the album.

The personal stuff is great.  Lean into it, it'll get more comfortable with time, and you'll both discover things about yourself, and potentially teach.  We're treating this as a "resume" (which it's good for), but it's also for you (which it will hopefully be great for).